HW1/P4/p4.py

```diff
@@ -0,0 +1,18 @@
+def main():
+	#CSV takes the file and creates and RDD of character,issue
+	csv=(sc.textFile('source.csv').map(lambda x: x.split('","'))
+	                                .map(lambda x:[x[0][1:],x[1][:-1]]))
+	csv.cache()
+	#issues_by_character groups the issues with character as key
+	issues_by_character=csv.groupByKey().map(lambda x:[x[0],list(x[1])])
+	#characters_by_issue creates a dict with issue as key and list of chars as value
+	characters_by_issue=csv.map(lambda x:[x[1],x[0]]).groupByKey().map(lambda x:[x[0],list(x[1])]).collectAsMap()
+	#linked characters is an RDD that has the x[0] as the char name and a list of chars its linked to as x[1]
```

Comment:
Isn't collectAsMap one of those expensive functions that we were not supposed to use?